drivers/postmanApiLoadDriver/postmanApi.go

```diff
@@ -0,0 +1,179 @@
+	podSpec := &corev1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: deploymentName + "-",
+			Namespace:    postmanParams.Namespace,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:    deploymentName,
+					Image:   postmanCentosImage,
+					Command: []string{"sleep", "infinity"},
+				},
+			},
+			RestartPolicy: corev1.RestartPolicyOnFailure,
+		},
+	}
+
+	pod, err := k8sCore.CreatePod(podSpec)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create pod [%s], Err: %v", podSpec.Name, err)
+	}
+
+	err = k8sCore.ValidatePod(pod, timeOut, timeInterval)
+	if err != nil {
+		return nil, fmt.Errorf("failed to validate pod [%s], Err: %v", pod.Name, err)
+	}
+	time.Sleep(1 * time.Minute)
+
+	return pod, nil
+}
+
+func RunKubectlCommandInPod(podName string, namespace string, kubeconfig string, command string) (string, error) {
+	cmd := fmt.Sprintf("kubectl --kubeconfig %v -n %v exec -it %v -- %v", kubeconfig, namespace, podName, command)
+	log.Infof("Command: ", cmd)
+	output, _, err := osutils.ExecShell(cmd)
+	if err != nil {
+		return "", err
+	}
+	log.Infof("Terminal output: %v", output)
+	return output, nil
+}
+
+func ExecuteCommandInShell(command string) (string, string, error) {
+	out, res, err := osutils.ExecShell(command)
+	if err != nil {
+		return "", "", err
+	}
+	return out, res, nil
+}
+
+func ExecutePostmanForPdsInAK8sPod(postmanParams *PostmanDriver) (*corev1.Pod, bool, error) {
+	collectionPath, err := GetPostmanCollectionPath()
+	if err != nil {
+		log.FailOnError(err, "Postman Collection Json not found, Please create a Collection json manually and export to {%v} folder", defaultCollectionPath)
+	}
+	log.InfoD("Postman Collection found is- %v", collectionPath)
+
+	postDep, err := postmanParams.RunNewmanWorkload(postmanParams)
+	copyCmd := "kubectl cp " + collectionPath + " " + postDep.Name + ":collection.json" + " -n " + postmanParams.Namespace
+	out, res, err := ExecuteCommandInShell(copyCmd)
+	if err != nil {
+		return nil, false, fmt.Errorf("there was some problem in executing Postman Newman container due to- [%v]", err)
+	}
+	log.Info("output of copy command is - %v, %v", out, res)
+	newmanCmd := "newman run collection.json --verbose"
+	output, err := RunKubectlCommandInPod(postDep.Name, postmanParams.Namespace, postmanParams.Kubeconfig, newmanCmd)
+	if err != nil {
+		return nil, false, fmt.Errorf("there was some problem in executing Postman Newman container due to- [%v]", err)
+	}
+	log.InfoD("output from the newman execution is- %v", output)
+	if strings.Contains(output, "failure") {
+		log.FailOnError(err, "newman exited with a failure.. Please check logs for more details")
+	}
+	k8sCore.DeletePod(postDep.Name, postDep.Namespace, true)
+	return postDep, true, nil
+}
+
+func ExecutePostmanCommandInTorpedo() (bool, error) {
+	collectionPath, err := GetPostmanCollectionPath()
+	if err != nil {
```

Comment:
Same comment - FailOnError checks internally - but you dont need it here. You should return err and handle it in the testcase